numpy/lib/format.py

```diff
@@ -265,20 +278,17 @@ def dtype_to_descr(dtype):
         replicate the input dtype.
 
     """
-    warn_msg = ("metadata on a dtype can be saved but cannot be read. Use "
-               "another form of storage for the array")
+    if _has_metadata(dtype):
+        warnings.warn("metadata on a dtype can be saved but will be ignored "
```

Comment:
Is it really true that the metadata is saved? This seems to imply a determined user can extract the metadata from the file by hand. My assumption was that the metadata was not saved at all.

Reply:
The dtypes in the test have a `fail` boolean, the one with True (which was the one reported in gh-14142) preserves the metadata, the rest remove it. Changing the message appropriately.

Reply:
Hmmm, I am not actually sure. The "dtype to descriptor for storing in npy file" function probably loses the metadata at least most of the time completely. So that would mean saving is currently possible and just a warning that it is lost is great.
I had tried actually saving it, but we use `saveeval` to load, and that can only succeed for super simple metadata.

Reply:
@seberg which part are you unsure about? I looked at the bytestring produced by the saving the different dtypes in the test, the ones that crash the loader are the ones that preserve the metadata, so sometimes it is preserved and sometimes not. When it is preserved, loading it crashes the loader.

Reply:
Yeah, sorry. I realized that a warning it must be, since sometimes we actually drop the metadata and sometimes we store. We could probably discriminate, but for now just "always warn" seems a fair start.